src/Runtime/include/OnnxMlir.h

```diff
@@ -0,0 +1,289 @@
+ */
+void *rmrGetData(RtMemRef *rmr);
+
+/**
+ * RtMemRef data setter
+ *
+ * @param rmr, pointer to the RtMemRef
+ * @param data, data buffer of the RtMemRef to be set
+ *
+ * Set the data buffer pointer of the RtMemRef. Note that the data buffer
+ * is assumed to be managed by the user, i.e., the RtMemRef destructor
+ * will not free the data buffer. Because we don't know how exactly the
+ * data buffer is allocated, e.g., it could have been allocated on the stack.
+ */
+void rmrSetData(RtMemRef *rmr, void *data);
+
+/**
+ * RtMemRef data sizes getter
+ *
+ * @param rmr, pointer to the RtMemRef
+ * @return pointer to the data shape array.
+ */
+INDEX_TYPE *rmrGetDataShape(RtMemRef *rmr);
+
+/**
+ * RtMemRef data sizes setter
+ *
+ * @param rmr, pointer to the RtMemRef
+ * @param dataSizes, data sizes array to be set
+ *
+ * Set the data sizes array of the RtMemRef to the values in the input array.
+ */
+void rmrSetDataShape(RtMemRef *rmr, INDEX_TYPE *dataSizes);
+
+/**
+ * RtMemRef data strides getter
+ *
+ * @param rmr, pointer to the RtMemRef
+ * @return pointer to the data strides array.
+ */
+int64_t *rmrGetDataStrides(RtMemRef *rmr);
+
+/**
+ * RtMemRef data strides setter
+ *
+ * @param rmr, pointer to the RtMemRef
+ * @param dataStrides, data strides array to be set
+ *
+ * Set the data strides array of the RtMemRef to the values in the input array.
+ */
+void rmrSetDataStrides(RtMemRef *rmr, int64_t *dataStrides);
+
+/**
+ * RtMemRef data type getter
+ *
+ * @param rmr, pointer to the RtMemRef
+ * @return ONNX data type of the data buffer elements.
+ */
+int rmrGetDataType(RtMemRef *rmr);
+
+/**
+ * RtMemRef data type setter
+ *
+ * @param rmr, pointer to the RtMemRef
+ * @param dataType, ONNX data type to be set
+ *
+ * Set the ONNX data type of the data buffer elements.
+ */
+void rmrSetDataType(RtMemRef *rmr, int dataType);
+
+/**
+ * RtMemRef data buffer size getter
+ *
+ * @param rmr, pointer to the RtMemRef
+ * @return the total size of the data buffer in bytes.
+ */
+// int64_t rmrGetDataBufferSize(RtMemRef *rmr);
+
+/**
+ * RtMemRef rank getter
+ *
+ * @param rmr, pointer to the RtMemRef
+ * @return rank of data sizes and strides of the RtMemRef.
+ */
+int rmrGetRank(RtMemRef *rmr);
+
+/**
+ * RtMemRef name getter
+ *
+ * @param rmr, pointer to the RtMemRef
+ * @return pointer to the name of the RtMemRef,
+ *         an empty string if the name is not set.
+ */
+char *rmrGetName(RtMemRef *rmr);
+
+/**
+ * RtMemRef name setter
+ *
+ * @param rmr, pointer to the RtMemRef
+ * @param name, name of the RtMemRef to be set
+ *
+ * Set the name of the RtMemRef.
+ */
+void rmrSetName(RtMemRef *rmr, char *name);
+
+/**
+ * RtMemRef number of elements getter
+ *
+ * @param rmr, pointer to the RtMemRef
+ * @return the number of elements in the data buffer.
+ */
+INDEX_TYPE rmrGetNumElems(RtMemRef *rmr);
+
+/*---------------------------------------- */
+/* C/C++ API for RtMemRefList calls */
+/*---------------------------------------- */
+
```

Comment:
please add a description of what the list is, not necessarily the fields but what it is intended for. This is where I would indicate that in this list, RtMemRefs can be accessed by index or names. Is it supposed to be consistent? If one name is provided, all must have names? 

Also define that list appears to be static in size.